File: OnlineResumeHosting/backend/src/main/java/com/example/resume/entity/VisitLog.java
package com.example.resume.entity;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "visit_log")
public class VisitLog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "resume_id", nullable = false)
    private Long resumeId;

    @Column(length = 50)
    private String ip;

    @Column(name = "user_agent", length = 500)
    private String userAgent;

    @Column(length = 500)
    private String referer;

    @CreationTimestamp
    @Column(name = "visited_at", updatable = false)
    private LocalDateTime visitedAt;
}

File: PersonalBlogCMS/api/entities/VisitLog.ts
import { Entity, Column } from 'typeorm';
import { Base } from './Base.js';

@Entity('visit_logs')
export class VisitLog extends Base {
  @Column({ name: 'article_id', nullable: true })
  articleId: number;

  @Column({ name: 'ip_address', length: 45, nullable: true })
  ipAddress: string;

  @Column({ name: 'user_agent', type: 'text', nullable: true })
  userAgent: string;

  @Column({ length: 255, nullable: true })
  referer: string;
}
